apps/platform-app/src/app/[locale]/(auth)/(components)/LinkedInLoginButton/LinkedInLoginButton.tsx

```diff
@@ -0,0 +1,32 @@
+'use client'
+import { AppRoutes } from '@/utils/routes'
+import { Button } from '@gdh/ui-system'
+import { GithubIcon } from '@gdh/ui-system/icons'
+import { signIn } from 'next-auth/react'
+import { useState } from 'react'
+import styles from '../GithubLoginButton/GithubLoginButton.module.scss'
+
+interface GithubLoginButtonProps {
+  label?: string
+}
+
+export const LinkedInLoginButton = (props: GithubLoginButtonProps) => {
+  const [isCalled, setIsCalled] = useState(false)
+  return (
+    <Button
+      disabled={isCalled}
+      onClick={() => {
+        setIsCalled(true)
+        signIn('linkedin', {
+          callbackUrl: AppRoutes.githubOAuth,
```

Comment:
**issue**: The callback URL is set to `AppRoutes.githubOAuth`. Ensure this is intentional for LinkedIn login, or update it to a LinkedIn-specific route.

Reply:
This is intentional. It's a general redirect route, and the name 'github-oauth' is now legacy and should be updated.

Reply:
if linkedin provider has to work for both specialists and hunters i have an idea to move role management from auth.ts to specific callback route, there will be one for specialist and one for hunters. i think this would be a better idea then managing roles during sign in process.

Reply:
good point